cmd/helper_server.go

```diff
@@ -226,3 +228,38 @@ func handlerFactories(keyManager rsakey.Manager) ([]proxy.Authenticator, []proxy
 			proxy.NewCredentialsIssuerCookies(),
 		}
 }
+
+func getTLSCertAndKey() (bool, tls.Certificate) {
+	var cert tls.Certificate
+	var tlsEnabled bool
+	var err error
+
+	tlsCert := viper.GetString("HTTP_TLS_CERT")
+	tlsKey := viper.GetString("HTTP_TLS_KEY")
+
+	if tlsCert == "" && tlsKey == "" {
```

Comment:
Can we improve this block? It does not have any tests and since it's difficult to read I'd love to see some reduction in complexity here. Also, I think that the naming of the variables is quite misleading and easy to misunderstand. A very simple way to improve this is with something like this:

```
var cert tls.Certificate
if certPath, keyPath := viper.GetString(), viper.GetString(); certPath == "" && keyPath == "" {
  cert, err := ...
  if err != nil ...
  return cert
}

if certString, keyString := viper.GetString(), viper.GetString(); certString== "" && keyString == "" {
  cert, err := ...
  if err != nil ...
  return cert
}
```

And then just check if `cert != nil` before `ListenAndServe`

Reply:
ok I'll do that. Please bear that the original section was terrible style, with variables shadowing at every corner... I'll try to add a test as well.

Reply:
My comment was not intended to be critical of you. I don't care who wrote the code - I write terrible code all the time. But while we work on this, let's improve it too! :)

Reply:
I used a boolean because this is a struct: the func cannot return nil, and testing that the returned value is an empty structure is ugly. However, I agree the if cascade if bad.

Reply:
You can return a pointer `*tls.Certificate` which can evaluate to `nil` instead

Reply:
why not. This just incurs some more changes in the existing code...